src/os/filestore/FileJournal.cc

```diff
@@ -1681,6 +1686,12 @@ void FileJournal::submit_entry(uint64_t seq, bufferlist& e, uint32_t orig_len,
     logger->inc(l_os_j_bytes, e.length());
   }
 
+  if (osd_op) {
+    osd_op->mark_event("commit_queued_for_journal_write");
+    osd_op->journal_trace.init("journal", &trace_endpoint, &osd_op->store_trace);
```

Comment:
this `journal_trace.init()` should be conditional on `if (osd_op->store_trace)` instead of just `if (osd_op)`, or we'll create separate traces that aren't attached to a parent

(the call to `osd_op->mark_event()` should remain outside of `if (osd_op->store_trace)`)